lib/awful/widget/layoutlist.lua

```diff
@@ -0,0 +1,442 @@
+        textbox:set_align(item.style.align or beautiful.layoutlist_align)
+    end
+
+    local text = ""
+
+    if item.name then
+        text = "<span color='"..fg.."'>"..item.name..'</span>'
+    end
+
+    return text, bg, nil, item.icon, {
+        shape               = shape,
+        shape_border_width  = shape_bw,
+        shape_border_color  = shape_bc,
+    }
+end
+
+module.source = {}
+
+--- The layout list for the first selected tag of a screen.
+-- @tparam screen s The screen.
+-- @sourcefunction awful.widget.layoutlist.source.for_screen
+function module.source.for_screen(s)
+    s = capi.screen[s or ascreen.focused() or 1]
+    assert(s)
+
+    local t = s.selected_tag or #s.tags > 0 and s.tags[1]
+
+    return t and t.layouts or {}
+end
+
+--- The layouts available for the first selected tag of `awful.screen.focussed()`.
+-- @sourcefunction awful.widget.layoutlist.source.current_screen
+function module.source.current_screen()
+    return module.source.for_screen()
+end
+
+--- The default layout list.
+-- @see awful.layout.layouts
+-- @sourcefunction awful.widget.layoutlist.source.default_layouts
+function module.source.default_layouts()
+    return alayout.layouts
+end
+
+local function reload_cache(self)
+    self._private.cache = {}
+
+    local show_text = (not self._private.style.disable_name) and
+        (not beautiful.layoutlist_disable_name)
+
+    local show_icon = (not self._private.style.disable_icon) and
+        (not beautiful.layoutlist_disable_icon)
+
+    local s = self.screen or ascreen.focused()
+
+    -- Get the list.
+    local ls = self:get_layouts()
+
+    for _, l in ipairs(ls or {}) do
+        local icn_path, icon = beautiful["layout_" .. (l.name or "")]
+        if icn_path then
+            icon = surface.load(icn_path)
+        end
+
+        table.insert(self._private.cache, {
+            icon         = show_icon and icon   or nil,
+            name         = show_text and l.name or nil,
+            layout       = l,
+            screen       = s,
+            callback     = function() alayout.set(l) end,
+            style        = self._private.style,
+        })
+    end
+end
+
+local function update(self)
+    assert(self._private.layout)
+    awcommon.list_update(
+        self._private.layout,
+        self._private.buttons or default_buttons,
+        wb_label,
+        self._private.data,
+        self._private.cache,
+        {
+            widget_template = self._private.widget_template
+        }
+    )
+end
+
+local layoutlist = {}
+
+--- The layoutlist default widget layout.
+-- If no layout is specified, a `wibox.layout.fixed.vertical` will be created
+-- automatically.
+-- @property base_layout
+-- @param widget
+-- @see wibox.layout.fixed.vertical
+-- @see base_layout
+
+--- The delegate widget template.
+-- @property widget_template
+-- @param table
+
+--- The layoutlist screen.
+-- @property screen
+-- @param screen
+
+--- A function that returns the list of layout to display.
+--
```

Comment:
Empty line. Can be removed?

Reply:
I don't know. It's rather common in let say, git, to have a newline after the summary. I agree we need some consistency and neither me or the other currently enforce it either way. Given that `ldoc` summary need to be 80 char max to fit in the CSS, at some point I may add a linter script to enforce this kind of ldoc policy.